### PR TITLE
DA API Certificate verification 

### PR DESCRIPTION
**Pull request by bacv**
Certificate verification in the mempool (follow-up to #634). Tests will be added as the Indexer service nears completion.

### CODE REVIEW

File: nomos-services/mempool/src/verify/mod.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+pub trait MempoolVerificationProvider {
+    type Payload;
+    type Parameters;
+    type Settings: Clone;
+
+    // TODO: Payload verification parameters most likely will come from another Overwatch service.
+    // Once it's decided, update the `new` method to require service relay as parameter.
+    fn new(settings: Self::Settings) -> Self;
+
+    async fn get_parameters(&self, payload: &Self::Payload) -> Self::Parameters;
```
Comment (danielSanchezQ) [inline nomos-services/mempool/src/verify/mod.rs:11]: Yes, this is my main concern. It is gonna be difficult to be generic over this settings if by settings we need to include relays to other services. The best way would be to change the fn signature to include an overwatch handler. But then we are introducing a dependency on overwatch that we may not want. We have to think about this. I have some other ideas like passing a channel that gets updated everytime the settings change (as they probably are not static, they may change over time). And then drop the overwatch into a `getter` async task. Probably not the best explanation neither.

TLDR: Lets go ahead with this, then change it further ahead when we see the flaws clearly.

File: nomos-services/mempool/src/verify/mod.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#[async_trait::async_trait]
+pub trait MempoolVerificationProvider {
```
Comment (danielSanchezQ) [inline nomos-services/mempool/src/verify/mod.rs:2]: If this is intended for `DaMempool` lets specify in the name. If its intended to be generic over mempools lets keep it.

**Review by danielSanchezQ — APPROVED** [review]
Another chunk of generics for the collection 🪙

File: nomos-services/mempool/src/verify/mod.rs (hunk 2 above)
Comment (bacv) [reply nomos-services/mempool/src/verify/mod.rs:2]: This definition is in `mempool/verify/mod.rs`, I'm pretty sure that we'll want some kind of verification for transactions? Da specific implementation is in `mempool/da/verify/fullreplication.rs`.

File: nomos-services/mempool/src/verify/mod.rs (hunk 1 above)
Comment (bacv) [reply nomos-services/mempool/src/verify/mod.rs:11]: In other services we have settings separated from the relay and I think it works well, this is example from mempool network adapter: https://github.com/logos-co/nomos-node/blob/69c88d94467e69b98a9227bc64fc2f618383a57e/nomos-services/mempool/src/network/mod.rs#L21-L24
I was mostly referring to this setup being missing in that TODO comment. But if you have an idea how we can improve this - lets discuss it.

Comment (danielSanchezQ) [reply nomos-services/mempool/src/verify/mod.rs:11]: Yes, but for other services settings are static. In the case for verification in our DA protocol, we need an updated list of registered DA nodes. That list comes from approved registrations in blocks, which changes over time (slots/epochs tbd). Which invalidates any static pre-configuration.
